adding cartocolors

andy-esch opened the pull request:
Porting over the schemes from the [CartoColors node module](https://github.com/CartoDB/cartocolor).

- [x] sequential: 2a1518b
- [x] sequential tests: finished in edad0e8
- [x] diverging: 2a1518b
- [x] diverging tests: finished in edad0e8
- [x] qualitative: 9215bbf
- [x] qualitative tests: aeeed6e
- [x] docs

andy-esch commented:
@jiffyclub, pretty much finished with the sequential and diverging palettes here. Any advice on handling qualitative palettes?

Also, looks like the py2.7 and pypy tests are failing because they're grabbing a too-new version of IPython. I've had to limit the IPython version in a setup.py to avoid this issue: https://github.com/CartoDB/cartoframes/blob/master/setup.py#L46-L48

jiffyclub commented:
Qualitative maps can be handled in pretty much the same way as the others (e.g. using the same class). For example in the colorbrewer maps I think the only difference between the qualitative palettes and the others is the palette type.

andy-esch commented:
@jiffyclub, could you do a code review of this? I modeled the new palette structure off of cmocean like you said. Only substantial change is adding a new named parameter to [`utils.get_map_factory`](https://github.com/jiffyclub/palettable/pull/21/files#diff-86b28c76eac01b4e1f98b8293cf09037R154) that controls how the colors are sampled from the palettes. In my case, the palettes are ordered in the way that the would ideal be sampled (e.g., first three would be a three color palette instead of first, middle, and last like the default).

Let me know if there's anything that you'd like me to change.

andy-esch commented:
@jiffyclub, any thing else you need from me here?

jiffyclub commented:
Thanks for this! I haven't had a chance to review in detail, but will try to get to it soon.

jiffyclub commented on palettable/utils.py:
```diff
@@ -190,8 +195,10 @@ def get_map(name, reverse=False):
+        if is_evenly_spaced:
+            colors = evenly_spaced_values(length, names_to_data[name])
+        else:
+            colors = names_to_data[name][0:length]
```
It looks like this PR is engineered so this will never have `length > len(names_to_data[name])`, but I wonder if there should be a check for future usage? Also, if you want, you can leave the `0` off this slice and do `colors = names_to_data[name][:length]`.

jiffyclub commented on palettable/utils.py:
```diff
@@ -189,9 +194,14 @@ def get_map(name, reverse=False):
 
         if name_lower not in name_map:
             raise KeyError('Unknown palette name: {0}'.format(name))
+        elif len(names_to_data[name]) < length:
```
👍 

Could you also add a test of this exception and the new `is_evenly_spaced` behavior in [test_utils.py](https://github.com/andy-esch/palettable/blob/5a6652d8db8fc3620f0e32ffd4843592b299a60c/palettable/test/test_utils.py)?

andy-esch commented:
Ok, I added the length check (and verified on my end if errors correctly if a palette is too short for ones requested to build), and changed the slice to same style as rest of code.

jiffyclub commented:
Awesome, thank you! Could you also add a test of the new `get_map_factory` feature and exception in test_utils.py?

andy-esch commented:
Yep :) here you go: a6452a1

jiffyclub commented:
Thanks, this looks good! I'm going to try to get all the tests passing in Travis before I merge this, but I expect to merge this soon.
